### PR TITLE
Add documentation for higher-level transaction API

### PR DESCRIPTION
**Pull request by lehni**
For a detailed discussion, see:

* https://github.com/strongloop/loopback-datasource-juggler/pull/1472
* https://github.com/strongloop/loopback-datasource-juggler/pull/1472#issuecomment-328463480

@bajtos please let me know if this looks good!

### CODE REVIEW

**Comment by lehni** [comment]
Hmm that's strange, why is @slnode doing this: 0bc605219ff9f095e5d872189937c3ef459b3647 ?

**Comment by crandmck** [comment]
@lehni This occurs because we have a Jenkins pipeline that runs [`update-readmes.sh`](https://github.com/strongloop/loopback.io/blob/gh-pages/update-readmes.sh) script
whenever there's a commit to the repo.  This script gets READMEs from various other repos to reuse them in the docs.  See http://loopback.io/doc/en/contrib/Including-READMEs.html for details.

In other words, don't be concerned.  It's normal and expected.

File: pages/en/lb3/Using-database-transactions.md [hunk 1]
```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
...
+  See [Lower-level Transaction API](#lower-level-transaction-api) below for more details.
+
+* To simplify and streamline the handling of transactions, [`dataSource.transaction()`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) offers a higher-level abstraction over the lower-level transaction API through a single method.  
+  See [Higher-level Transaction API](#higher-level-transaction-api) below for more details.
+
+## Higher-level Transaction API
+
+Inspired by the way [Objection.js](https://github.com/Vincit/objection.js) is handling transactions by [binding them to models](http://vincit.github.io/objection.js/#binding-models-to-a-transaction), the method [`dataSource.transaction(execute, options, cb)`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) offers an interface into a simple way to create transactions and bind them to models, so that the transaction object never needs to be explicitly created, passed along to Model database methods, or committed / rolled-back.
```
Comment (crandmck) [inline pages/en/lb3/Using-database-transactions.md]: Minor wording suggestion: 

"Inspired by how [Objection.js](https://github.com/Vincit/objection.js) handles transactions, the method [`dataSource.transaction(execute, options, cb)`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) creates transactions and binds them to models.  Thus, you don't need to explicitly create a transaction object, pass it along to Model database methods, nor committed / rolled-back.

File: pages/en/lb3/Using-database-transactions.md [hunk 2]
```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
...
+
+Calling the `transaction(execute, options, cb)` method instead creates the transaction for you, then calls the passed `execute` function if it is provided and passes a `models` argument to it containing all models in this data-source, but as versions that are automatically bound to the transaction that was just created. Smart caching is used to only created the bound models for the properties of `models` that are actually requested.
+
+At the end, `transaction.commit()` or `transaction.rollback()` are then automatically called, based on whether exceptions happen during execution or not. If no callback is provided to be called at the end of the execution, a Promise object is returned that is resolved or rejected as soon as the execution is completed, and the transaction is committed or rolled back.
+
+### Examples
+
+Here is how to create an successfully commit a transaction:
```
Comment (crandmck) [inline pages/en/lb3/Using-database-transactions.md]: Should this be "... create and commit a transaction:"  ?? (I don't think it's necessary to say "successfully").

File: pages/en/lb3/Using-database-transactions.md [hunk 3]
```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
...
+}
+console.log(await app.models.MyModel.count()); // 0
+```
+
+### Options
+
+The same options are supported as by the `options` argument of the lower-level [`beginTransaction()`](http://apidocs.loopback.io/loopback-datasource-juggler/#transactionmixin-begintransaction) method:
+
+* `otions.isolationLevel`: See [Isolation levels](#isolation-levels)
+* `otions.timeout`: See [Timeout Handling](#timeout-handling)
```
Comment (crandmck) [inline pages/en/lb3/Using-database-transactions.md]: Should these be `options.isolationLevel` and `options.timeout` ?

**Review by crandmck — CHANGES_REQUESTED** [review]

File: pages/en/lb3/Using-database-transactions.md (hunk 3 above)
Comment (lehni) [reply pages/en/lb3/Using-database-transactions.md]: oops, indeed! 😬

File: pages/en/lb3/Using-database-transactions.md (hunk 1 above)
Comment (lehni) [reply pages/en/lb3/Using-database-transactions.md]: Agreed, but I think I'll link to the specific page from the Objection.js link: http://vincit.github.io/objection.js/#binding-models-to-a-transaction

**Comment by lehni** [comment]
@crandmck I've just addressed the points raised in your review. Thanks!

**Comment by crandmck** [comment]
LGTM. 
@bajtos PTAL

File: pages/en/lb3/Using-database-transactions.md [hunk 4]
```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
...
+### Examples
+
+Here is how to create and commit a transaction:
+```js
+await app.dataSources.db.transaction(async models => {
+  const {MyModel} = models;
+  console.log(await MyModel.count()); // 0
+  await MyModel.create({foo: 'bar'});
+  console.log(await MyModel.count()); // 1
+})
```
Comment (bajtos) [inline pages/en/lb3/Using-database-transactions.md]: Nitpick: missing semicolon.

**Review by bajtos — APPROVED** [review]
This is great, thank you @lehni!

File: pages/en/lb3/Using-database-transactions.md (hunk 4 above)
Comment (lehni) [reply pages/en/lb3/Using-database-transactions.md]: Oops! I never write them, forgot that one. Will fix now. Shall I merge after?

Comment (bajtos) [reply pages/en/lb3/Using-database-transactions.md]: > Shall I merge after?

Yes please. Just squash the commits into a single one please, and wait for green CI results.

**Comment by crandmck** [comment]
Thanks @lehni !

**Pull request merged**